### PR TITLE
Feature/fbfw 28 mui dialog

### PR DESCRIPTION
**Pull request by sariacodazen**
Dialog ready.

Testing Plan: Use WDModal component inside the react root to render it. Then click on button to see modal appear. Test in different viewports.

### CODE REVIEW

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          sx={{
+            p: 2,
+            background: "linear-gradient(to right, white 96%, transparent 4%)",
+            maxWidth: "57vw",
+          }}
+        >
+          <div className="modal__content">
+            Are you sure you want to exit the game?
+          </div>
+        </Typography>
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:80]: lines 70 - 80 should be children passed to this component. however, you will need to wrap them with the modal__content div.

File: beta-src/src/App.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import * as React from "react";
 import "./assets/css/App.css";
 import Box from "@mui/material/Box";
 import map from "./assets/svg/map.svg";
+import WDModal from "./components/wd-modal/WDModal";
```
Comment (nebevets) [inline beta-src/src/App.tsx:5]: we don't want to import this in App. this PR is about the component build, not implementation of the component

File: beta-src/src/assets/css/App.css [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/* Components CSS Files */
+@import url("../../components/wd-modal/WDModal.css");
```
Comment (nebevets) [inline beta-src/src/assets/css/App.css:2]: ideally, we should incorporate component specific css in the component itself.

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            "&::before": {
+              background: "linear-gradient(45deg, transparent 50%, white 50%)",
+              content: '""',
+              display: "block",
+              position: "absolute",
+              width: 22,
+              height: 22,
+              top: 10,
+              transform: "rotate(45deg)",
+              right: 5,
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:66]: this css should be alphabetized. also, if we can move this to variable declared outside the component it would be better. it makes the component cleaner and puts all the css in one place.

```
const arrowStyles = {
            position: "relative",
            mt: "0px",
            "&::before": {
              background: "linear-gradient(45deg, transparent 50%, white 50%)",
           ...
};

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            },
+          }}
+        />
+        <Typography
+          sx={{
+            p: 2,
+            background: "linear-gradient(to right, white 96%, transparent 4%)",
+            maxWidth: "57vw",
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:74]: let's make a variable for this.

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        transformOrigin={{
+          vertical: "center",
+          horizontal: "right",
+        }}
+        PaperProps={{
+          style: {
+            backgroundColor: "transparent",
+            boxShadow: "none",
+            borderRadius: 0,
+          },
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:47]: let's make variable for this, abc properties.

**Review by nebevets — CHANGES_REQUESTED** [review]
add a more detailed test plan that shows an implementation with adding that implementation to the codebase. have a video and/or screenshots that show how the modal looks and works. have a link to the figma for the modal for comparison purposes. nice progress!

**Comment by sariacodazen** [comment]
TEST PLAN: Import WDModal component into App and render it to view. The container is absolute so the component user can determine positioning of modal container later on. The purpose of this task is to create a working modal that points to its trigger, and can take different content for the dialog, and for the icon trigger.

Figma: https://www.figma.com/file/JnpO52R22kG3lB2wexs9rB/webDiplomacy
 
Screen Shot 2022-02-02 at 10 38 00 AM
Screen Shot 2022-02-02 at 10 38 07 AM
Screen Shot 2022-02-02 at 10 38 18 AM
Screen Shot 2022-02-02 at 10 38 22 AM

**Review by hyukcodazen — APPROVED** [review]
overall LGTM. In my opinion, we can separate the modal activation logic and let parent handle it. In case we want this modal to be used for different purpose like warnings and popups.

**Comment by sariacodazen** [comment]
NEW BOX SHADOW AND RADIUS
Screen Shot 2022-02-02 at 12 53 09 PM
S

**Review by bryancodazen — CHANGES_REQUESTED** [review]
.

File: beta-src/package-lock.json [hunk 7]
```diff
@@ -12,6 +12,7 @@
         "@emotion/styled": "^11.6.0",
         "@mui/icons-material": "^5.3.0",
         "@mui/material": "^5.3.0",
+        "@mui/system": "^5.3.0",
```
Comment (nebevets) [inline beta-src/package-lock.json:15]: this still need to remove this

File: beta-src/package.json [hunk 8]
```diff
@@ -8,6 +8,7 @@
     "@emotion/styled": "^11.6.0",
     "@mui/icons-material": "^5.3.0",
     "@mui/material": "^5.3.0",
+    "@mui/system": "^5.3.0",
```
Comment (nebevets) [inline beta-src/package.json:11]: remove

File: beta-src/package-lock.json [hunk 9]
```diff
@@ -38185,13 +38186,13 @@
       }
     },
     "prop-types": {
-      "version": "15.7.2",
-      "resolved": "https://registry.npmjs.org/prop-types/-/prop-types-15.7.2.tgz",
-      "integrity": "sha512-8QQikdH7//R2vurIJSutZ1smHYTcLpRWEOlHnzcWHmBYrOGUysKwSsrC89BCiFj3CbrfJ/nXFdJepOVrY1GCHQ==",
+      "version": "15.8.1",
```
Comment (nebevets) [inline beta-src/package-lock.json:38189]: this file should not be part of this PR

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+const WDModal: React.FC<WDModalProps> = function ({
+  ariaDescribedBy,
+  children,
+  rightPositioning,
+  topPositioning,
+  triggerIcon,
+}) {
+  const anchorEl = useRef(null);
+
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:22]: remove space here

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+  };
+
+  return (
+    <div
+      className="modal__container"
+      style={{ top: `${topPositioning}%`, right: `${rightPositioning}%` }}
+    >
+      <div className="modal__trigger-container">
+        <Button
+          aria-describedby={ariaDescribedBy}
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:40]: alpha order all your props on every component.

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+              height: 22,
+              position: "absolute",
+              right: 5,
+              top: 10,
+              transform: "rotate(45deg)",
+              width: 22,
+            },
+          }}
+        />
+
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:87]: remove this space

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+  };
+
+  const handleClose = () => {
+    setIsOpen(false);
+  };
+
+  return (
+    <div
+      className="modal__container"
+      style={{ top: `${topPositioning}%`, right: `${rightPositioning}%` }}
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:36]: i think we're going to have a positioning component to handle this, so you don't need this for this PR

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+  const handleOpen = () => {
+    setIsOpen(true);
+  };
+
+  const handleClose = () => {
+    setIsOpen(false);
+  };
+
+  return (
+    <div
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:34]: we should use Box instead of div. This allows you to use sx

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+              position: "absolute",
+              right: 5,
+              top: 10,
+              transform: "rotate(45deg)",
+              width: 22,
+            },
+          }}
+        />
+
+        <div className="modal__content">{children}</div>
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:88]: make this a box and ditch the imported CSS classes

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+      </Popover>
+    </div>
+  );
+};
+
+export default WDModal;
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:94]: let's change the name to WDPopover

File: beta-src/src/components/wd-modal/WDModal.tsx [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+    setIsOpen(false);
+  };
+
+  return (
+    <div
+      className="modal__container"
+      style={{ top: `${topPositioning}%`, right: `${rightPositioning}%` }}
+    >
+      <div className="modal__trigger-container">
+        <Button
```
Comment (nebevets) [inline beta-src/src/components/wd-modal/WDModal.tsx:39]: after discussion with bryan, i think we should use an unstyled button component for this...this way we can pass either an icon or text for the button. https://mui.com/api/button-unstyled/

File: beta-src/src/components/wd-popover/WDPopover.tsx [hunk 18]
```diff
@@ -0,0 +1,87 @@
+import * as React from "react";
+import { useRef } from "react";
+import { Popover, Box } from "@mui/material";
```
Comment (nebevets) [inline beta-src/src/components/wd-popover/WDPopover.tsx:3]: import { Box, Popover } from "@mui/material";

alpha order whenever possible.

File: beta-src/src/components/wd-popover/WDPopover.tsx [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+
+interface WDPopoverProps {
+  children: React.ReactNode;
+  handleClose: (event: React.MouseEvent<HTMLButtonElement>) => void; // A function that sets isOpen to false
```
Comment (nebevets) [inline beta-src/src/components/wd-popover/WDPopover.tsx:7]: let's add the comments above each prop like so:
```
/**
 * A function that sets isOpen to false
 * /
handleClose: (event: React.MouseEvent<HTMLButtonElement>) => void; 
```

let's change the comment to:

A callback that is called when the Popover closes

File: beta-src/src/components/wd-popover/WDPopover.tsx [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+  isOpen: boolean;
+  trigger: React.ReactNode; // The popover trigger; must have an onClick callback attached that sets isOpen to true
```
Comment (nebevets) [inline beta-src/src/components/wd-popover/WDPopover.tsx:9]: same as above on comment styling.

let's rename this prop to `popoverElement` with a description of:

"A component that opens or closes the Popover when clicked"

File: beta-src/src/components/wd-popover/WDPopover.tsx [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+        transformOrigin={{
+          vertical: "center",
+          horizontal: "right",
+        }}
+        PaperProps={{
+          style: {
+            backgroundColor: "transparent",
+            boxShadow: "black -5px 4px 6px -4px",
+            borderRadius: "3px 13px 13px 3px",
+          },
```
Comment (nebevets) [inline beta-src/src/components/wd-popover/WDPopover.tsx:51]: alpha order all these props on Popover and PaperProps styles

File: beta-src/src/components/wd-popover/WDPopover.tsx [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+          horizontal: "left",
+        }}
+        transformOrigin={{
+          vertical: "center",
+          horizontal: "right",
+        }}
+        PaperProps={{
+          style: {
+            backgroundColor: "transparent",
+            boxShadow: "black -5px 4px 6px -4px",
```
Comment (nebevets) [inline beta-src/src/components/wd-popover/WDPopover.tsx:49]: the color comes last.

File: beta-src/src/components/wd-popover/WDPopover.tsx [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+          pt: "15px",
+        }}
+        ref={anchorEl}
+      >
+        {trigger}
+      </Box>
+      <Popover
+        open={isOpen}
+        anchorEl={anchorEl.current}
+        onClose={handleClose}
```
Comment (nebevets) [inline beta-src/src/components/wd-popover/WDPopover.tsx:37]: let's make `handleClose` `onClose` instead.

File: beta-src/src/components/wd-popover/WDPopover.tsx [hunk 24]
```diff
@@ -0,0 +1,87 @@
...
+          },
+        }}
+      >
+        <Box
+          sx={{
+            mt: "0px",
+            position: "relative",
+            "&::before": {
+              background: "linear-gradient(45deg, transparent 50%, white 50%)",
+              content: '""',
```
Comment (nebevets) [inline beta-src/src/components/wd-popover/WDPopover.tsx:60]: why double quotes here? i think it should just be content: ""

File: beta-src/src/components/wd-popover/WDPopover.tsx [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+              display: "block",
```
Comment (nebevets) [inline beta-src/src/components/wd-popover/WDPopover.tsx:61]: not sure we need to set display i think it's block be default

File: beta-src/src/components/wd-popover/WDPopover.tsx [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+              transform: "rotate(45deg)",
+              width: 22,
+            },
+          }}
+        />
+
+        <Box
+          sx={{
+            background: "linear-gradient(to right, white 94%, transparent 6%)",
+            maxWidth: "57vw",
```
Comment (nebevets) [inline beta-src/src/components/wd-popover/WDPopover.tsx:75]: are we sure about this width? pls check the figma and confirm this. if we have to use breakpoints and fixed widths that is fine.

File: beta-src/src/components/wd-popover/WDPopover.tsx [hunk 27]
```diff
@@ -0,0 +1,87 @@
+import * as React from "react";
```
Comment (nebevets) [inline beta-src/src/components/wd-popover/WDPopover.tsx:1]: let's move this file outside the wd-popover folder into components pls.

**Review by nebevets — COMMENTED** [review]
almost there! just some really nit picky stuff now. please address and we're good to go

File: beta-src/src/components/wd-popover/WDPopover.tsx (hunk 24 above)
Comment (sariacodazen) [reply beta-src/src/components/wd-popover/WDPopover.tsx:60]: It breaks with out it, needed for MUI

File: beta-src/src/components/WDPopover.tsx [hunk 28]
```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+import { useRef } from "react";
+import { Box, Popover } from "@mui/material";
+
+interface WDPopoverProps {
+  children: React.ReactNode;
+  isOpen: boolean;
+  /**
+   *A callback that sets isOpen to false
```
Comment (nebevets) [inline beta-src/src/components/WDPopover.tsx:9]: it may not do that. this callback could do anything. we need to just say:

a callback to be used when the popover closes

or something to that effect.

File: beta-src/src/components/WDPopover.tsx [hunk 29]
```diff
@@ -0,0 +1,91 @@
...
+        {popoverTrigger}
+      </Box>
+      <Popover
+        anchorEl={anchorEl.current}
+        anchorOrigin={{
+          horizontal: "left",
+          vertical: "center",
+        }}
+        open={isOpen}
+        onClose={onClose}
```
Comment (nebevets) [inline beta-src/src/components/WDPopover.tsx:47]: onClose, the open

File: beta-src/src/components/WDPopover.tsx [hunk 30]
```diff
@@ -0,0 +1,91 @@
...
+          vertical: "center",
+        }}
+      >
+        <Box
+          sx={{
+            mt: "0px",
+            position: "relative",
+            "&::before": {
+              background: "linear-gradient(45deg, transparent 50%, white 50%)",
+              content: '""',
```
Comment (nebevets) [inline beta-src/src/components/WDPopover.tsx:66]: why are there two sets of quotes?

File: beta-src/src/components/WDPopover.tsx [hunk 31]
```diff
@@ -0,0 +1,91 @@
...
+import { useRef } from "react";
+import { Box, Popover } from "@mui/material";
+
+interface WDPopoverProps {
+  children: React.ReactNode;
+  isOpen: boolean;
+  /**
+   *A callback that sets isOpen to false
+   */
+  onClose: (event: React.MouseEvent<HTMLButtonElement>) => void; // A function that sets isOpen to false
```
Comment (nebevets) [inline beta-src/src/components/WDPopover.tsx:11]: remove the // comment at the end

**Review by nebevets — APPROVED** [review]
LGTM. please get bryan's final approval. nice work

File: beta-src/src/components/WDPopover.tsx [hunk 32]
```diff
@@ -0,0 +1,91 @@
...
+import { useRef } from "react";
+import { Box, Popover } from "@mui/material";
+
+interface WDPopoverProps {
+  children: React.ReactNode;
+  isOpen: boolean;
+  /**
+   * A callback to be used when the popover closes
+   */
+  onClose: (event: React.MouseEvent<HTMLButtonElement>) => void;
```
Comment (bryancodazen) [inline beta-src/src/components/WDPopover.tsx:11]: This needs to be optional, you shouldn't be forced to pass a callback when the popover closes.

**Review by bryancodazen — CHANGES_REQUESTED** [review]
Awesome work!

Before we merge this, make sure to re-generate the files into the `/beta` directory and check that no errors occur when using the docker harness.

File: beta-src/src/components/WDPopover.tsx (hunk 32 above)
Comment (sariacodazen) [reply beta-src/src/components/WDPopover.tsx:11]: So we decided to control modal functionality from parent so we need to pass that handler to make isOpen false

Comment (nebevets) [reply beta-src/src/components/WDPopover.tsx:11]: @bryancodazen this onClose is required by popover...we are just passing it through and allowing the parent of this component to control that close behavior.

Comment (bryancodazen) [reply beta-src/src/components/WDPopover.tsx:11]: @sariacodazen @nebevets let's sync up on this tomorrow (02/08) during the 15 min standup time if possible.

**Review by bryancodazen — APPROVED** [review]
LGTM
